src/avro_util.erl

```diff
@@ -295,6 +296,113 @@ resolve_duplicated_refs(Type0) ->
   {Type, _Refs} = resolve_duplicated_refs(Type0, []),
   Type.
 
+%% @doc Check if writer schema is compatible to reader schema.
+-spec is_compatible(avro_type(), avro_type()) -> true | {not_compatible, _, _}.
```

Comment:
Also, would be nice to have typespecs for `RPath`/`WPath` at least to understand what to expect.